shared/src/main/scala/kinesis4cats/producer/fs2/FS2Producer.scala

```diff
@@ -124,20 +126,39 @@ abstract class FS2Producer[F[_], PutReq, PutRes](implicit
                 )
               } yield ()
             }
-            .onError { case e: Throwable => deferredError.complete(e).void }
         }
         .compile
         .drain
     } yield ()
   }
 
-  private[kinesis4cats] def resource: Resource[F, Unit] = for {
-    deferredError <- Deferred[F, Throwable].toResource
-    _ <- Resource
-      .make(start(deferredError).start)(stop)
-      .void
-      .race(deferredError.get.flatMap(F.raiseError[Unit]).toResource)
-  } yield ()
+  private[kinesis4cats] def resource: Resource[F, Unit] = {
+    def retryCtx(details: retry.RetryDetails) =
+      LogContext().addRetryDetails(details).context
+    val loop =
+      retry.retryingOnFailuresAndAllErrors(
```

Comment:
I'm not sure that i want a retry loop at this level. I think this is logic that users would want to define as needed. Some users may prefer a fatal exception, or to do something else with the errors. 

I think it would make sense to add the logs if the loop dies, but we should remove the retry loop from this area while also fixing the Resource start.

Reply:
I think the background loop retry has to be here? The producer loop is an internal detail of producer (not exposed otherwise), currently no other way to interact with it. 

Made configurable via `FS2Producer.Config`, with a default `Capped(5.seconds, Backoff(100.millis))`.

The `F[F[Unit]]` from #164 is great, that gives us the chance to observe f.e the size of internal queue which I wanted to introduce on our side via callbacks, but this is even better as it's with per per record granularity. 

The problem is, if the background loop dies for good (because for example, user overrode the `RetryPolicy` to `alwaysGiveUp`), it won't be signalled at all via `F[F[Unit]]` mechanism.

I made small change to this PR to close channel if the loop dies (before only cleaned up as part of resource). This should be enough to signal that the user cannot send messages to the channel and return error immediately.  Also, since closing the channel does not cancel already blocked `send` requests, raced it against `channel.closed` as described in `Channel.close` docs.

If this PR is accepted in conjunction with #164, we could complete all `Deferred` instances from the `Channel` too, to signal problem that way too.